### ISSUE TITLE
[BUG]: Accessibility checker reports 4.1.2 violation while it seems correct

### ISSUE THREAD

**Issue by gcattan** [IBMa/equal-access#759]
### Project

accessibility-checker-engine, ruleId: Rpt_Aria_RequiredParent_Native_Host_Sematics

### Description

Hello, 

The accessibility checker is finding a bug with the carbon-react-component `Tab`:
![image](https://user-images.githubusercontent.com/6229031/159684045-82d42df3-9bb2-4e2d-9686-e85d03ccf6d3.png)

However, when I look at the element, I saw that the element with `tab` role is a child of the `ul` element with `role="tablist"`
![image](https://user-images.githubusercontent.com/6229031/159683944-f981fd69-0956-4951-8196-00b93d9839f8.png)

### Steps to reproduce

Steps for reproducing need investigation. 

Thank you in advance for any help!

**Comment by tombrunet** [comment]
Which archive / ruleset are you using? I cannot reproduce this error using `latest` or `preview` on https://react.carbondesignsystem.com/iframe.html?id=components-tabs--default&args=&viewMode=story

**Comment by gcattan** [comment]
Thank for digging into it! I am using the `Latest - Latest Deployment.`. It is very strange I cannot reproduce the error in the sandbox too. 
I made a couple of test too, and the strangest I observed is that if I do a pretty print of the html, the error is gone ^^

**Comment by gcattan** [comment]
I think the problem appear because the `Tabs` are locate inside a `div` with `role="button"`.
In simple html, if I remove the role of the div, the error is gone:

![image](https://user-images.githubusercontent.com/6229031/159878798-634fa6f7-a337-43da-a192-98ecb69b9689.png)

**Comment by tombrunet** [comment]
https://codesandbox.io/s/issue-759-b06oou

**Comment by shunguoy** [comment]
the original test case has a button inside a button.  A button element is a singular element and should contain another button.  The checker should report this, rather than escalate to a different issue.  Also, see related  #643

**Comment by marcjohlic** [comment]
Phill will investigate this one to see what we need to do.

**Comment by shunguoy** [comment]
This issue can happen for many elements.  #1027 lists the elements with 14 roles that can potentially cause the same or similar issues and [ARIA in HTML](https://w3c.github.io/html-aria/#allowed-descendants-of-aria-roles) specifies "Allowed descendants of ARIA roles" for all the ARIA roles.  Therefore, this issue needs to be addressed in a generic way in order to fix all related issues.  
    1.  A potential new rule that checks if a descendant is valid for a given element role according to [ARIA in HTML](https://w3c.github.io/html-aria/#allowed-descendants-of-aria-roles).
    2.  what to do if a descendant is invalid. #1027 handle a special case for the element roles with "Presentational Children: true".
    3. Other invalid descendant cases (to be investigated).

**Comment by fbarroso24** [comment]
@shunguoy so is this a false positive or is this a different type of violation?  If its still a violation, does it still fall under the 4.1.2 category or a new category?  Thanks.

**Comment by shunguoy** [comment]
@fbarroso24  we are planning to modify the rules to report certain usage errors, for example, using a focusable element as a child of an element with a 'button' role

**Comment by fbarroso24** [comment]
@shunguoy I see.  So it sounds like this is still a violation.  Would this still fall under the [4.1.2: Name, Role, Value](https://www.ibm.com/able/requirements/requirements/#4_1_2) wcag guideline?

**Comment by shunguoy** [comment]
@fbarroso24  we will have a discussion with SMEs to decide this as if it should be a new rule or covered in the current rule. I don't have a specific time frame for this either. Thanks for your patience on this.

**Comment by MHoov** [comment]
@shunguoy has the SME discussion occurred yet?

**Comment by shunguoy** [comment]
The fix aims to reduce false positives and overlaps among different rules. The following rules need to be updated as the result of the new rule "aria_descendant_valid":

 aria_child_valid:

 IBMA_Focus_MultiTab

 IBMA_Focus_Tabbable

 Rpt_Aria_RequiredParent_Native_Host_Sematics

 test case updates. About 50 test cases need to be re-tested because of the rule update. Potentially, new test cases need to be created for the rules.

 baseline updates The selenium and karma baseline needs to be updated for the changes.

**Comment by eirckr** [comment]
This seems to be working now, @gcattan can you please re-test this on our next release with preview rules. The new release should be available later this week.
